### PR TITLE
not ready, 10 from 12 tests passed

### PR DESCRIPTION
**Pull request by valeriitovstyk**

### CODE REVIEW

File: src/deepEqual.js [hunk 1]
```diff
@@ -22,9 +22,41 @@
...
+    }
+    return true;
+  } else {
+    return false;
+  }
 }
 
 module.exports = deepEqual;
+
+deepEqual({ test: 5, extra: 123 }, { test: 5, extra: null });
```
Comment (roman-mate) [inline src/deepEqual.js]: remove all of the lines below.

File: src/deepEqual.js [hunk 2]
```diff
@@ -22,9 +22,41 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  *
  */
-
 function deepEqual(a, b) {
-  // write code here
+  if (a === b) {
+    return a === b;
```
Comment (roman-mate) [inline src/deepEqual.js]: just "return true;"

File: src/deepEqual.js [hunk 3]
```diff
@@ -22,9 +22,41 @@
...
+  } else if ((typeof a === typeof b && typeof a === 'object' && a !== null && b !== null)) {
```
Comment (roman-mate) [inline src/deepEqual.js]: you can remove else. That will reduce level of nesting in this function.

File: src/deepEqual.js [hunk 4]
```diff
@@ -22,9 +22,41 @@
...
+    }
+
+    for (let property of Object.keys(a)) {
+      if (b.hasOwnProperty(property)) {
+        if (!deepEqual(a[property], b[property])) {
+          return false;
+        }
+      } else {
+        return false;
+      }
```
Comment (roman-mate) [inline src/deepEqual.js]: simplify this into one if.

File: src/deepEqual.js (hunk 1 above)
Comment (valeriitovstyk) [reply src/deepEqual.js]: done

File: src/deepEqual.js (hunk 2 above)
Comment (valeriitovstyk) [reply src/deepEqual.js]: done

File: src/deepEqual.js (hunk 3 above)
Comment (valeriitovstyk) [reply src/deepEqual.js]: done

File: src/deepEqual.js (hunk 4 above)
Comment (valeriitovstyk) [reply src/deepEqual.js]: done

File: src/deepEqual.js [hunk 5]
```diff
@@ -22,9 +22,25 @@
...
+    return true;
+  }
+  if ((typeof a === typeof b && typeof a === 'object' && a !== null && b !== null)) {
+    if (Object.keys(a).length !== Object.keys(b).length) {
+      return false;
+    }
+
+    for (let property of Object.keys(a)) {
+      if (b.hasOwnProperty(property)) {
+        if (!deepEqual(a[property], b[property])) {
```
Comment (roman-mate) [inline src/deepEqual.js]:
```suggestion
        if (b.hasOwnProperty(property) && !deepEqual(a[property], b[property])) {
```
